lib/librato/metrics/aggregator.rb

```diff
@@ -52,20 +52,33 @@ def initialize(opts={})
       # @return [Aggregator] returns self
       def add(measurements)
         measurements.each do |metric, data|
+          entry = {}
           if @prefix
             metric = "#{@prefix}.#{metric}"
           end
+          entry[:name] = metric.to_s
           if data.respond_to?(:each) # hash form
+            validate_parameters(data)
             value = data[:value]
             if data[:source]
-              metric = "#{metric}#{SOURCE_SEPARATOR}#{data[:source]}"
+              metric = "#{metric}#{SEPARATOR}#{data[:source]}"
+              entry[:source] = data[:source].to_s
+            end
+            if data[:tags] && data[:tags].respond_to?(:each)
+              metric = metric.to_s
+              data[:tags].sort.each do |key, value|
```

Comment:
Since building a hash key from a set of tags occurs a few times in `librato-rack` as well, perhaps this is better suited for a util module?